### DISCUSSION TITLE
Support for ESLint's new FlatConfig

### DISCUSSION THREAD

**Discussion by Nyaacinth** [zed-industries/zed#7204] [Feature Requests]
FlatConfig: https://eslint.org/docs/latest/use/configure/configuration-files-new

I recently started using Zed, it looks really promising, but now I found a small blocker on my way: seems like Zed doesn't support ESLint's new config format. ESLint announced this feature about half or 1 years ago, and now it's feature completed and able to use. It contains several improvements like `export default`(ESM) instead of `module.exports=`(CJS) and the ability to use different configs for different files through regex matching.

It would be really nice to see Zed with support for this. (Or if I'm wrong, then glad to see how to enable it)

**Comment by SomeoneToIgnore** [comment]
For eslint, Zed starts an LSP server and passes a set of parameters to it 
https://github.com/zed-industries/zed/blob/0102ffbfcab7fd6f74474e46b8e8b1746daa1041/crates/zed/src/languages/typescript.rs#L203

apart from that, there's no code in Zed that somehow configures eslint, and all work is on the LSP server.
Slightly more flexible approach Zed has is prettier, which uses prettier CLI via small LSP wrapper managed by Zed: https://github.com/zed-industries/zed/blob/0102ffbfcab7fd6f74474e46b8e8b1746daa1041/crates/prettier/src/prettier.rs#L39
it allows passing through certain Zed configuration items and dealing with various configuration options via prettier CLI logic.

**Reply by Nyaacinth** [reply]
After some tries to figure out, I found this

https://github.com/zed-industries/zed/blob/8bafc61ef57f8f1807ced6f8b69891c4ee701a7d/crates/zed/src/languages/typescript.rs#L217-L232

It perfectly match config schema of vscode-eslint, seems like we can somehow add a config entry that can add `"experimental": { "useFlatConfig": true }` here to enable FlatConfig support on vscode-eslint which is the language server Zed is using.

**Reply by motine** [reply]
did you ever solve this?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support

**Reply by motine** [reply]
hey @JosephTLyons, thanks for the clarification. it would be great if the results of such support inquiries could be documented somewhere. unfortunately, i can not google for answers if they are in the discord. i would like to avoid bothering the team with my questions. wdyt?
